argilla/src/argilla/records/_resource.py

```diff
@@ -258,27 +262,62 @@ def from_model(cls, model: RecordModel, dataset: "Dataset") -> "Record":
         )
 
 
-class RecordFields:
+class RecordFields(dict):
     """This is a container class for the fields of a Record.
-    It allows for accessing fields by attribute and iterating over them.
+    It allows for accessing fields by attribute and key name.
     """
 
-    def __init__(self, fields: Dict[str, Union[str, None]]) -> None:
-        self.__fields = fields or {}
-        for key, value in self.__fields.items():
-            setattr(self, key, value)
+    def __init__(self, fields: Optional[Dict[str, FieldValue]] = None) -> None:
+        super().__init__(fields or {})
 
-    def __getitem__(self, key: str) -> Optional[str]:
-        return self.__fields.get(key)
+    def __getattr__(self, item: str):
+        return self[item]
 
-    def __iter__(self):
-        return iter(self.__fields)
+    def __setattr__(self, key: str, value: MetadataValue):
+        self[key] = value
 
-    def to_dict(self) -> Dict[str, Union[str, None]]:
-        return self.__fields
+    def to_dict(self) -> dict:
+        return {key: value for key, value in self.items()}
 
-    def __repr__(self) -> str:
-        return self.to_dict().__repr__()
+
+class RecordMetadata(dict):
+    """This is a container class for the metadata of a Record."""
+
+    def __init__(self, metadata: Optional[Dict[str, MetadataValue]] = None) -> None:
+        super().__init__(metadata or {})
+
+    def __getattr__(self, item: str):
```

Comment:
Are we definitely going to support both of these strategies?
```python
record.vectors["new-vector"] = [1.0, 2.0, 3.0]
record.vectors.vector = [1.0, 2.0, 3.0]

record.metadata["new-key"] = "new_value"
record.metadata.key = "new_value"
```

Personally, I'm not 100% sure about the attribute based approach because I don't like how the IDE behaves. Is there a way to improve the experience there?

Reply:
Ok. So we'll open a new PR for suggestions and responses, where we'll use a dictionary style access as well? Nice.